### DISCUSSION TITLE
Integration with other email providers, especially self-hosted

### DISCUSSION THREAD

**Discussion by lukas-mertens** [twentyhq/twenty#5489] [Ideas]
## Scope & Context

The ticket is about the gmail integration/email integration in general

## Current behavior

Currently it is only possible to integrate with gmail as it seems

## Expected behavior
I am part of an organization that runs most if its infrastructure self-hosted, including its email server. Twenty would be a great addition as a CRM. Currently it is not possible to integrate with self hosted email solutions however. An integration with mailcow/roundcube or simply imap (to allow integration with most email providers) would be great, so that not only gmail is supported.

**Comment by ajojotank** [comment]
I agree with this, although I am not sure on how to implement it as I am not a developer.

**Comment by FelixMalfait** [comment]
Moving to discussion. This is a highly requested feature. We need to continue working on the basics before we extend to other email providers but it will come at some point for sure!

**Comment by olmohake** [comment]
I am currently looking into different os crms for a non-profit organisation. Was really exited about twenty. However, being able to sync with exiting mail accounts is an hard requirement for us as well.

**Comment by jpstrikesback** [comment]
Also looking at twenty for a NPO, and an email fw/cc/bcc listener inbox (see #5807) is a requirement and would perhaps circumvent or delay the need for direct integrations (while losing the ability to send from twenty), I think a listener inbox would provide a more secure base option that gets the job done for more orgs.

**Reply by tfbecker** [reply]
+1

**Comment by nser77** [comment]
Same problem here.

We also need to to distribute those workers.


Is it possible?

**Comment by lhjt** [comment]
Also interested in seeing twenty support email providers outside of G oogle. My suggestion would be to support IMAP/SMTP directly, instead of having direct integrations with specific mailbox providers (and similarly use CalDAV for calendar scheduling).

This would mean that twenty could work with essentially every email provider as my understanding is the majority of providers support these protocols.

**Comment by chsasank** [comment]
Code seems to be organized in a way that they are open for other messaging platforms: https://github.com/twentyhq/twenty/blob/258fd078394c95da0e5790c59a5ec1352e120ec7/packages/twenty-server/src/modules/messaging/message-import-manager/drivers/gmail/providers/gmail-client.provider.ts

**Comment by chsasank** [comment]
A work around for the timebeing can probably be found by creating a free gmail ID and integrating it that with your email: https://support.google.com/mail/answer/21289?sjid=5968582876535744111-AP

**Comment by kbkozlev** [comment]
I have the same issue, perhaps connecting the emails using IMAP or POP3 could be a solution

**Comment by vinyll** [comment]
I would also need a raw IMAP, SMTP and caldav support for business account

**Comment by talhakaanozkan** [comment]
I think it would be the right decision to bring SMTP - IMAP, POP3 support.

**Comment by folliesandjollies** [comment]
Any updates on the progress of this feature? Thanks!

**Comment by FelixMalfait** [comment]
Hey unfortunately, this hasn't been started yet. If someone wants to work on this we're open to paying a bounty for it.
(Need to be someone experienced/senior as this is quite a big task)

**Reply by zhyd1997** [reply]
Hi

I have experience with IMAP and SMTP, and I will look into it!

**Reply by scradeley** [reply]
What kind of bounty? How would that work?

**Reply by severinwullschleger** [reply]
hi @FelixMalfait, we started working on it on our fork. what is the bounty for bringing it upstream? I'm asking because it is probably quite some extra work because we are relatively far behind from upstream. We plan to merge your upstream anyways at some point but currently we have higher priorities on our timeline, that's why I need to plan ahead a bit

**Reply by FelixMalfait** [reply]
Hey @severinwullschleger thanks! There's a PR in review that's going to be merged soon: https://github.com/twentyhq/twenty/pull/12576

**Comment by zhyd1997** [comment]
I recommend using the SendGrid service to send emails.

**Reply by FelixMalfait** [reply]
No, people want to be free to use whatever email provider they already have.

**Reply by zhyd1997** [reply]
It should work, and SandGrid emails will not got to spam folder as it promises.

**Reply by gonzaleztroyano** [reply]
As @FelixMalfait stated, it's nice to have SendGrid as an option, but having an **standard SMTP** option will be **much better**, as you can configure whatever email provider you already have 🙂

**Reply by zhyd1997** [reply]
Good idea!

**Reply by lukas-mertens** [reply]
I think you are both misunderstanding the problem. This is not only about sending, but mainly about receiving mails and integrating them. The CRM needs both ingoing and outgoing emails so IMAP is probably the correct protocol, not SMTP. SMTP would be nice in a second iteration to also allow sending emails directly from Twenty, but most important is IMAP in my opinion.

**Reply by zhyd1997** [reply]
@lukas-mertens  Yes, you can not only send but also receive emails:

https://github.com/zhyd1997/twenty-mail

**Reply by lukas-mertens** [reply]
@zhyd1997 I don't understand how the repository you linked is related to this problem? This just looks like an AI generated example on how to send and receive emails in node.js which is not the major problem of an issue like this. The main problem is integrating it into the CRM and mapping the existing functionality to SMTP + IMAP. 

The best way to contribute to this as a non-senior-developer would probably be (maintainers, correct me if I am wrong) to make some sketches e.g. in excalidraw on how SMTP/IMAP Support could be integrated into the existing UI. This could be a great basis for discussion and it makes it clearer for everyone which fields have to be added to APIs etc.

Another thing someone more technical could do is to go through IMAP/SMTP spec and validate that all the functionality we have is indeed possible to just map to the other protocol

**Reply by zhyd1997** [reply]
I see, the repo is just a demo, I have experience with similar tasks for CRM (1. integrate Outlook for project A and 2. set up email webhook and parse email content for project B) in my previous work experience.

Anyway, someone may have taken the task.

**Comment by FelixMalfait** [comment]
Imap is now done thanks to @neo773 - it's behind a feature flag available in the lab (only on `main` for now, the next tagged image should come next week). He's working on smtp/caldav next which should also come soon!

**Reply by amrit-Cheeema** [reply]
any progress on this?

**Reply by zacnoo** [reply]
> any progress on this?

I believe this is the PR, if you wanted to subscribe to it.

https://github.com/twentyhq/twenty/pull/12993

**Comment by Novapixel1010** [comment]
I noticed both PR #12993 and #12576  is merged. So what version should this be on or is there more testing/work to be done? I am on version `1.8.2` and don't see anything for setting this up? I am happy to bug test this if need.

**Reply by FelixMalfait** [reply]
<img width="950" height="701" alt="Screenshot 2025-10-20 at 17 02 11" src="https://github.com/user-attachments/assets/712d1008-81b5-4ab1-aff3-b62e53c39de3" />
It's still in the lab but I believe @neo773 will get it out soon!

**Reply by FelixMalfait** [reply]
In the meantime you can enable it manually

**Reply by Novapixel1010** [reply]
Thanks
